pkg/e2e/infra/utils.go

```diff
@@ -118,6 +126,79 @@ func getMachineSetList(client runtimeclient.Client) (*mapiv1beta1.MachineSetList
 	return &machineSetList, nil
 }
 
+// getMachineSetList returns a MachineSetList object filtered by machineRoleLabel
+func getWorkerMachineSetList(client runtimeclient.Client) (*mapiv1beta1.MachineSetList, error) {
```

Comment:
Unless we need access to either `TypeMeta` or `ListMeta` why not return the slice (i.e., `[]MachineSet`). 

And ditto for the other functions that return `<Foo>List`.